sky/provision/aws/instance.py

```diff
@@ -1,17 +1,30 @@
 """AWS instance provisioning."""
+import re
+import time
 from typing import Any, Dict, List, Optional
 
 from botocore import config
+from botocore import exceptions
```

Comment:
We should use `sky.adaptors` instead of directly importing these library, to avoid these libraries being imported with `import sky`